lib/cdo/user_helpers.rb

```diff
@@ -7,29 +7,39 @@ module UserHelpers
   USERNAME_ALLOWED_CHARACTERS = /[a-z0-9\-\_\.]/
 
   def self.generate_username(queryable, name)
-    prefix = name.downcase.gsub(/[^#{USERNAME_ALLOWED_CHARACTERS.source}]+/, ' ')[0..16].squish.tr(' ', '_')
+    prefix = name.downcase.
+      gsub(/[^#{USERNAME_ALLOWED_CHARACTERS.source}]+/, ' ')[0..16].
+      squish.
+      tr(' ', '_')
 
     if prefix.empty? || prefix == ''
       prefix = 'coder' + (rand(900000) + 100000).to_s
     end
-
     prefix = "coder_#{prefix}" if prefix.length < 5
 
     return prefix if queryable.where(username: prefix).limit(1).empty?
 
+    # Throw darts to find an appropriate suffix, using it if we hit bullseye.
+    (0..1).each do |exponent|
```

Comment:
I'd suggest going up to 4 digits. For a sense of scale of duplicates, right now we have 667 users in production with the name 'John Smith'. So throwing darts at only up to 3 digits might fall back to the slow way fairly frequently. Going up to 4 digits would make it unlikely to fall back to the slow way. There doesn't seem to be reason to go higher than 4.